### PR TITLE
Apply new GraphQl Field structure to UI

### PR DESCRIPTION
**Pull request by ymoskovko-epam**
Jira Ticket [HW-52231](https://jira.site1.hyperwallet.local/browse/HW-52231)

**!!! cartfile content was temporary changed**. Original is `github "hyperwallet/hyperwallet-ios-sdk" "1.0.0-beta02"`

### CODE REVIEW

File: Sources/Localization/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -64,14 +65,18 @@
 // Add Transfer Method
 "create_account_label" = "Create Account";
 "enter_your_information_formatter" = "Enter your %@ information";
-"account_header" = "%1$@ - %2$@ (%3$@)";
 "bank_account_footer" = "On a paper check, the routing number is printed as the first 9 digits on the bottom of the check, followed by 4-17 digits for the account number.";
 "bank_card_footer" = "CVV is a 3 digit number, typically found on the back of the card.";
 "paypal_account_footer" = "PayPal transfer method email address should be same as profile email address.";
-"add_profile_footer" = "Note: we are not able to support adding an account for someone else.";
-"profile_header"="Account Holder";
+"account_holder_footer" = "Note: we are not able to support adding an account for someone else.";
```
Comment (jsaini-hw) [inline Sources/Localization/en.lproj/Localizable.strings]: We have removed footer from the code. Footer for now will have only error message if any. No constant text. So you can remove this footer

Comment (ymoskovko-epam) [reply Sources/Localization/en.lproj/Localizable.strings]: fixed this one

File: Sources/TransferMethod/AddTransferMethodViewController.swift [hunk 2]
```diff
@@ -379,51 +379,45 @@ extension AddTransferMethodViewController: AddTransferMethodView {
...
+                cells: newWidgets
+            )
+            presenter.sections.append(section)
+            widgets.append(contentsOf: newWidgets)
         }
     }
 
-    private func addInfoSection(_ transferMethodTypeDetail: TransferMethodTypeDetail) {
-        guard transferMethodTypeDetail.fees != nil || transferMethodTypeDetail.processingTime != nil else {
+    private func addInfoSection() {
```
Comment (jsaini-hw) [inline Sources/TransferMethod/AddTransferMethodViewController.swift]: Update func signature to ``` addInfoSection(_ transferMethodType: HyperwalletTransferMethodType) ```

File: Sources/TransferMethod/SelectTransferMethodTypePresenter.swift [hunk 3]
```diff
@@ -30,7 +30,7 @@ protocol SelectTransferMethodTypeView: class {
     func navigateToAddTransferMethodController(country: String,
                                                currency: String,
                                                profileType: String,
-                                               detail: TransferMethodTypeDetail)
+                                               transferMethodType: HyperwalletTransferMethodType)
```
Comment (jsaini-hw) [inline Sources/TransferMethod/SelectTransferMethodTypePresenter.swift]: Update to transferMethodTypeCode: String

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift [hunk 4]
```diff
@@ -16,11 +16,12 @@
 // DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
+import HyperwalletSDK
 import UIKit
 
 /// Represents the country and currency data to be displyed on the CountryCurrencyCell
 struct SelectTransferMethodTypeConfiguration {
-    let transferMethodType: String
+    let transferMethodType: HyperwalletTransferMethodType
```
Comment (jsaini-hw) [inline Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: This should still be String. let transferMethodTypeName: String

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift [hunk 5]
```diff
@@ -71,9 +72,9 @@ final class SelectTransferMethodTypeCell: UITableViewCell {
 
 extension SelectTransferMethodTypeCell {
     func configure(configuration: SelectTransferMethodTypeConfiguration) {
-        accessibilityIdentifier = configuration.transferMethodType
+        accessibilityIdentifier = configuration.transferMethodType.code
```
Comment (jsaini-hw) [inline Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: use configuration.transferMethodTypeName

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift [hunk 6]
```diff
@@ -71,9 +72,9 @@ final class SelectTransferMethodTypeCell: UITableViewCell {
...
 
-        textLabel?.text = configuration.transferMethodType.localized()
+        textLabel?.text = configuration.transferMethodType.name
```
Comment (jsaini-hw) [inline Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: use configuration.transferMethodTypeName

File: Sources/TransferMethod/SelectTransferMethodTypePresenter.swift [hunk 7]
```diff
@@ -65,13 +65,14 @@ final class SelectTransferMethodTypePresenter {
...
-
-        return SelectTransferMethodTypeConfiguration(transferMethodType: detail.transferMethodType.lowercased(),
-                                                     feesProcessingTime: feesProcessingTime,
-                                                     transferMethodIconFont: transferMethodIcon)
+        let transferMethodType = transferMethodTypes[index]
+        let feesProcessingTime = transferMethodType.formatFeesProcessingTime()
+        let transferMethodIcon = HyperwalletIcon.of(transferMethodType.code!).rawValue
+
+        return SelectTransferMethodTypeConfiguration(
+            transferMethodType: transferMethodType,
```
Comment (jsaini-hw) [inline Sources/TransferMethod/SelectTransferMethodTypePresenter.swift]: transferMethodType.name

File: Sources/TransferMethod/SelectTransferMethodTypeViewController.swift [hunk 8]
```diff
@@ -123,11 +123,11 @@ extension SelectTransferMethodTypeViewController: SelectTransferMethodTypeView {
     func navigateToAddTransferMethodController(country: String,
                                                currency: String,
                                                profileType: String,
-                                               detail: TransferMethodTypeDetail) {
+                                               transferMethodType: HyperwalletTransferMethodType) {
         let addTransferMethodController = AddTransferMethodViewController(country,
                                                                           currency,
                                                                           profileType,
-                                                                          detail.transferMethodType)
+                                                                          transferMethodType)
```
Comment (jsaini-hw) [inline Sources/TransferMethod/SelectTransferMethodTypeViewController.swift]: transferMethodType.code

File: Sources/TransferMethod/SelectTransferMethodTypeViewController.swift [hunk 9]
```diff
@@ -123,11 +123,11 @@ extension SelectTransferMethodTypeViewController: SelectTransferMethodTypeView {
     func navigateToAddTransferMethodController(country: String,
                                                currency: String,
                                                profileType: String,
-                                               detail: TransferMethodTypeDetail) {
+                                               transferMethodType: HyperwalletTransferMethodType) {
```
Comment (jsaini-hw) [inline Sources/TransferMethod/SelectTransferMethodTypeViewController.swift]: transferMethodType: String

File: Sources/TransferMethod/AddTransferMethodPresenter.swift [hunk 10]
```diff
@@ -76,16 +78,14 @@ final class AddTransferMethodPresenter {
                             strongSelf.loadTransferMethodConfigurationFields() })
                         return
                     }
-
-                    guard let result = result else {
-                        return
+                    guard
+                        let result = result,
+                        let fields = result.fieldGroups(),
```
[the remark this answers is not in the document]
Comment (ymoskovko-epam) [reply Sources/TransferMethod/AddTransferMethodPresenter.swift]: fixed

File: Sources/TransferMethod/AddTransferMethodPresenter.swift [hunk 11]
```diff
@@ -76,16 +76,13 @@ final class AddTransferMethodPresenter {
                             strongSelf.loadTransferMethodConfigurationFields() })
                         return
                     }
-
-                    guard let result = result else {
-                        return
+                    guard
+                        let result = result,
+                        let fields = result.fieldGroups()
```
[the remark this answers is not in the document]
Comment (ymoskovko-epam) [reply Sources/TransferMethod/AddTransferMethodPresenter.swift]: fixed

File: Sources/TransferMethod/AddTransferMethodViewController.swift [hunk 12]
```diff
@@ -40,7 +40,7 @@ public final class AddTransferMethodViewController: UITableViewController {
     private var country: String
     private var currency: String
     private var profileType: String
-    private var transferMethodType: String
+    private var transferMethodType: HyperwalletTransferMethodType
```
[the remark this answers is not in the document]
Comment (ymoskovko-epam) [reply Sources/TransferMethod/AddTransferMethodViewController.swift]: fixed

File: Sources/TransferMethod/AddTransferMethodViewController.swift [hunk 13]
```diff
@@ -93,7 +93,7 @@ public final class AddTransferMethodViewController: UITableViewController {
     public init(_ country: String,
                 _ currency: String,
                 _ profileType: String,
-                _ transferMethodType: String) {
+                _ transferMethodType: HyperwalletTransferMethodType) {
```
[the remark this answers is not in the document]
Comment (ymoskovko-epam) [reply Sources/TransferMethod/AddTransferMethodViewController.swift]: fixed

File: Sources/HyperwalletUI.swift [hunk 14]
```diff
@@ -68,10 +68,11 @@ public final class HyperwalletUI {
...
     /// - Returns: An instance of `AddTransferMethodViewController`
-    public func addTransferMethodViewController(_ country: String,
-                                                _ currency: String,
-                                                _ profileType: String,
-                                                _ transferMethodType: String) -> AddTransferMethodViewController {
+    public func addTransferMethodViewController(
+        _ country: String,
+        _ currency: String,
+        _ profileType: String,
+        _ transferMethodType: HyperwalletTransferMethodType) -> AddTransferMethodViewController {
```
[the remark this answers is not in the document]
Comment (ymoskovko-epam) [reply Sources/HyperwalletUI.swift]: fixed

File: Sources/TransferMethod/AddTransferMethodViewController.swift [hunk 15]
```diff
@@ -108,7 +108,7 @@ public final class AddTransferMethodViewController: UITableViewController {
 
     override public func viewDidLoad() {
         super.viewDidLoad()
-        title = transferMethodType.lowercased().localized()
+        title = transferMethodType.name
```
[the remark this answers is not in the document]
Comment (ymoskovko-epam) [reply Sources/TransferMethod/AddTransferMethodViewController.swift]: fixed

File: Sources/TransferMethod/AddTransferMethodViewController.swift [hunk 16]
```diff
@@ -307,9 +307,9 @@ extension AddTransferMethodViewController: AddTransferMethodView {
         }
     }
 
-    func showTransferMethodFields(_ fields: [HyperwalletField], _ transferMethodTypeDetail: TransferMethodTypeDetail) {
-        addFieldsSection(fields)
-        addInfoSection(transferMethodTypeDetail)
+    func showTransferMethodFields(_ fieldGroups: [HyperwalletFieldGroup]) {
```
[the remark this answers is not in the document]
Comment (ymoskovko-epam) [reply Sources/TransferMethod/AddTransferMethodViewController.swift]: fixed

File: Sources/TransferMethod/AddTransferMethodViewController.swift [hunk 17]
```diff
@@ -307,9 +307,9 @@ extension AddTransferMethodViewController: AddTransferMethodView {
...
+        addFieldsSection(fieldGroups)
+        addInfoSection()
```
[the remark this answers is not in the document]
Comment (ymoskovko-epam) [reply Sources/TransferMethod/AddTransferMethodViewController.swift]: fixed

File: Sources/TransferMethod/AddTransferMethodViewController.swift [hunk 18]
```diff
@@ -145,7 +145,7 @@ public final class AddTransferMethodViewController: UITableViewController {
                                                country,
                                                currency,
                                                profileType,
-                                               transferMethodType)
+                                               transferMethodType.code)
```
[the remark this answers is not in the document]
Comment (ymoskovko-epam) [reply Sources/TransferMethod/AddTransferMethodViewController.swift]: fixed

File: Sources/TransferMethod/AddTransferMethodViewController.swift (hunk 2 above)
Comment (ymoskovko-epam) [reply Sources/TransferMethod/AddTransferMethodViewController.swift]: fixed

File: Sources/TransferMethod/SelectTransferMethodTypePresenter.swift (hunk 3 above)
Comment (ymoskovko-epam) [reply Sources/TransferMethod/SelectTransferMethodTypePresenter.swift]: fixed

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift (hunk 4 above)
Comment (ymoskovko-epam) [reply Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: fixed

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift (hunk 5 above)
Comment (ymoskovko-epam) [reply Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: fixed

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift (hunk 6 above)
Comment (ymoskovko-epam) [reply Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: fixed

File: Sources/TransferMethod/SelectTransferMethodTypePresenter.swift (hunk 7 above)
Comment (ymoskovko-epam) [reply Sources/TransferMethod/SelectTransferMethodTypePresenter.swift]: fixed

File: Sources/TransferMethod/SelectTransferMethodTypeViewController.swift (hunk 9 above)
Comment (ymoskovko-epam) [reply Sources/TransferMethod/SelectTransferMethodTypeViewController.swift]: fixed

File: Sources/TransferMethod/SelectTransferMethodTypeViewController.swift (hunk 8 above)
Comment (ymoskovko-epam) [reply Sources/TransferMethod/SelectTransferMethodTypeViewController.swift]: fixed

File: Cartfile [hunk 19]
```diff
@@ -1 +1 @@
-github "hyperwallet/hyperwallet-ios-sdk" "1.0.0-beta02"
+github "hyperwallet/hyperwallet-ios-sdk" "task/HW-52223-Implement-New-GraphQl-Key-Data-Structures"
```
Comment (jsaini-hw) [inline Cartfile]: change it development

Comment (bhordiienko-epam) [reply Cartfile]: > change it development

done

**Pull request merged**